<h1> Palmer Penguin Data Analytics</h1>
<hr>
<h3> Rommel Gio C. Bermudez </h3>

In [1]:
import numpy as np
import pandas as pd

In [2]:
penguins = pd.read_csv('dataset//penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [4]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h4>Cleaning the Data Frame</h4>

In [5]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h5>Selecting a Column</h5>
<hr>Sytax
data_frame(column_name)

In [36]:
penguins[penguins ['island']=='Dream' ]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [37]:
penguins ['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: category
Categories (3, object): ['Adelie', 'Chinstrap', 'Gentoo']

In [38]:
A = penguins[penguins ['species'] == 'Adelie']

In [39]:
B = penguins[penguins ['species'] == 'Chinstrap']

In [40]:
C = penguins[penguins ['species'] == 'Gentoo']

In [41]:
A

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen        39.10000       18.70000         181.000000   
1    Adelie  Torgersen        39.50000       17.40000         186.000000   
2    Adelie  Torgersen        40.30000       18.00000         195.000000   
3    Adelie  Torgersen        43.92193       17.15117         200.915205   
4    Adelie  Torgersen        36.70000       19.30000         193.000000   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream        36.60000       18.40000         184.000000   
148  Adelie      Dream        36.00000       17.80000         195.000000   
149  Adelie      Dream        37.80000       18.10000         193.000000   
150  Adelie      Dream        36.00000       17.10000         187.000000   
151  Adelie      Dream        41.50000       18.50000         201.000000   

     body_mass_g     sex  year  
0    3750.000000    male  2007  
1    3800.000000  female  2007  
2    3250.000000  female  2007  
3    4201.754386    male  2007  
4    3450.000000  female  2007  
..           ...     ...   ...  
147  3475.000000  female  2009  
148  3450.000000  female  2009  
149  3750.000000    male  2009  
150  3700.000000  female  2009  
151  4000.000000    male  2009  

[152 rows x 8 columns]

In [42]:
B

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [43]:
C

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe        46.10000       13.20000         211.000000   
153  Gentoo  Biscoe        50.00000       16.30000         230.000000   
154  Gentoo  Biscoe        48.70000       14.10000         210.000000   
155  Gentoo  Biscoe        50.00000       15.20000         218.000000   
156  Gentoo  Biscoe        47.60000       14.50000         215.000000   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year  
152  4500.000000  female  2007  
153  5700.000000    male  2007  
154  4450.000000  female  2007  
155  5700.000000    male  2007  
156  5400.000000    male  2007  
..           ...     ...   ...  
271  4201.754386    male  2009  
272  4850.000000  female  2009  
273  5750.000000    male  2009  
274  5200.000000  female  2009  
275  5400.000000    male  2009  

[124 rows x 8 columns]

In [44]:
penguins ['body_mass_g'] >= 4000

0      False
1      False
2      False
3       True
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [45]:
(penguins ['species'] == 'Chinstrap') & (penguins ['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [46]:
penguins[(penguins ['species'] == 'Chinstrap') & (penguins ['body_mass_g'] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [47]:
datapenfemale1 = penguins[(penguins ['species'] == 'Chinstrap') & (penguins ['sex'] == 'female') & (penguins ['body_mass_g'] >= 4000)]

In [48]:
datapenmale1 = penguins[(penguins ['species'] == 'Chinstrap') & (penguins ['sex'] == 'male') & (penguins ['body_mass_g'] >= 4000)]

In [49]:
datapenmale1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 15 entries, 289 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            15 non-null     category
 1   island             15 non-null     category
 2   bill_length_mm     15 non-null     float64 
 3   bill_depth_mm      15 non-null     float64 
 4   flipper_length_mm  15 non-null     float64 
 5   body_mass_g        15 non-null     float64 
 6   sex                15 non-null     category
 7   year               15 non-null     category
dtypes: category(4), float64(4)
memory usage: 1.2 KB


In [50]:
datapenfemale1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1 entries, 284 to 284
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            1 non-null      category
 1   island             1 non-null      category
 2   bill_length_mm     1 non-null      float64 
 3   bill_depth_mm      1 non-null      float64 
 4   flipper_length_mm  1 non-null      float64 
 5   body_mass_g        1 non-null      float64 
 6   sex                1 non-null      category
 7   year               1 non-null      category
dtypes: category(4), float64(4)
memory usage: 564.0 bytes


In [51]:
penguins['bill_length_mm'] = penguins ['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [52]:
penguins['bill_depth_mm'] = penguins ['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [53]:
penguins['flipper_length_mm'] = penguins ['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [54]:
penguins['body_mass_g'] = penguins ['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [55]:
penguins['sex'] = penguins ['sex'].fillna(penguins['sex'].mode()[0])

In [56]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2> How to Change the values <h2>

In [57]:
penguins ['species'] = penguins ['species'].astype('category')

In [58]:
penguins ['island'] = penguins ['island'].astype('category')

In [59]:
penguins ['sex'] = penguins ['sex'].astype('category')

In [60]:
penguins ['year'] = penguins ['year'].astype('category')

In [61]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<h1>Exploratory Data Analytics

<h2>1.Which of the penguins have more Bill Depth

In [62]:
penguins[penguins ['bill_depth_mm'] >= 21]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
13  Adelie  Torgersen            38.6           21.2              191.0   
14  Adelie  Torgersen            34.6           21.1              198.0   
19  Adelie  Torgersen            46.0           21.5              194.0   
35  Adelie      Dream            39.2           21.1              196.0   
49  Adelie      Dream            42.3           21.2              191.0   
61  Adelie     Biscoe            41.3           21.1              195.0   

    body_mass_g   sex  year  
13       3800.0  male  2007  
14       4400.0  male  2007  
19       4200.0  male  2007  
35       4150.0  male  2007  
49       4150.0  male  2007  
61       4400.0  male  2008

<h2>Insight#1
<h3>It was the Adelie species from island torgesen ID 19 having the most bill length

<h2>2.which of the penguins have the bigger bill length

In [63]:
penguins[penguins ['bill_length_mm'] >= 55]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
185     Gentoo  Biscoe            59.6           17.0              230.0   
253     Gentoo  Biscoe            55.9           17.0              228.0   
267     Gentoo  Biscoe            55.1           16.0              230.0   
293  Chinstrap   Dream            58.0           17.8              181.0   
339  Chinstrap   Dream            55.8           19.8              207.0   

     body_mass_g     sex  year  
185       6050.0    male  2007  
253       5600.0    male  2009  
267       5850.0    male  2009  
293       3700.0  female  2007  
339       4000.0    male  2009

<h2>Insight#2
<h3>The one that has the Longest Bill length is penguin 185 from biscoe species is Gentoo with 59.6 mm

<h2>3.which penguin has the most flipper length

In [64]:
penguins[penguins ['flipper_length_mm'] >= 230]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
153  Gentoo  Biscoe            50.0           16.3              230.0   
185  Gentoo  Biscoe            59.6           17.0              230.0   
215  Gentoo  Biscoe            54.3           15.7              231.0   
217  Gentoo  Biscoe            49.8           16.8              230.0   
227  Gentoo  Biscoe            48.6           16.0              230.0   
241  Gentoo  Biscoe            52.1           17.0              230.0   
265  Gentoo  Biscoe            51.5           16.3              230.0   
267  Gentoo  Biscoe            55.1           16.0              230.0   

     body_mass_g   sex  year  
153       5700.0  male  2007  
185       6050.0  male  2007  
215       5650.0  male  2008  
217       5700.0  male  2008  
227       5800.0  male  2008  
241       5550.0  male  2009  
265       5500.0  male  2009  
267       5850.0  male  2009

<h2>Insight#3
<h3>The penguin that has the most Longest flipper is from gentoo species Island Biscoe ID:215

<h2>4.Which Penguin is the lightest in body mass

In [68]:
penguins[penguins ['body_mass_g'] <= 2900]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
54      Adelie     Biscoe            34.5           18.1              187.0   
58      Adelie     Biscoe            36.5           16.6              181.0   
64      Adelie     Biscoe            36.4           17.1              184.0   
98      Adelie      Dream            33.1           16.1              178.0   
116     Adelie  Torgersen            38.6           17.0              188.0   
298  Chinstrap      Dream            43.2           16.6              187.0   
314  Chinstrap      Dream            46.9           16.6              192.0   

     body_mass_g     sex  year  
54        2900.0  female  2008  
58        2850.0  female  2008  
64        2850.0  female  2008  
98        2900.0  female  2008  
116       2900.0  female  2009  
298       2900.0  female  2007  
314       2700.0  female  2008

<h2>Insight#4
<h3>The penguin thats the lightest is a penguin thats a chinstrap Island Dream her ID is: 314

<h2>5.Penguins that have the oldest recorded data year

In [74]:
penguins[penguins ['year'] == 2007]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
297  Chinstrap      Dream        48.50000       17.50000         191.000000   
298  Chinstrap      Dream        43.20000       16.60000         187.000000   
299  Chinstrap      Dream        50.60000       19.40000         193.000000   
300  Chinstrap      Dream        46.70000       17.90000         195.000000   
301  Chinstrap      Dream        52.00000       19.00000         197.000000   

     body_mass_g     sex  year  
0    3750.000000    male  2007  
1    3800.000000  female  2007  
2    3250.000000  female  2007  
3    4201.754386    male  2007  
4    3450.000000  female  2007  
..           ...     ...   ...  
297  3400.000000    male  2007  
298  2900.000000  female  2007  
299  3800.000000    male  2007  
300  3300.000000  female  2007  
301  4150.000000    male  2007  

[110 rows x 8 columns]

<h2>Insights#5
<h3>from penguins ID 0 to 301 is from 2007 and 2007 is the oldest

<h2>6.How many male and females penguins are there

In [86]:
penguins [penguins ['sex'] == 'female']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
4       Adelie  Torgersen            36.7           19.3              193.0   
6       Adelie  Torgersen            38.9           17.8              181.0   
12      Adelie  Torgersen            41.1           17.6              182.0   
..         ...        ...             ...            ...                ...   
335  Chinstrap      Dream            45.6           19.4              194.0   
337  Chinstrap      Dream            46.8           16.5              189.0   
338  Chinstrap      Dream            45.7           17.0              195.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
1         3800.0  female  2007  
2         3250.0  female  2007  
4         3450.0  female  2007  
6         3625.0  female  2007  
12        3200.0  female  2007  
..           ...     ...   ...  
335       3525.0  female  2009  
337       3650.0  female  2009  
338       3650.0  female  2009  
340       3400.0  female  2009  
343       3775.0  female  2009  

[165 rows x 8 columns]

In [87]:
penguins [penguins ['sex'] == 'male']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
5       Adelie  Torgersen        39.30000       20.60000         190.000000   
7       Adelie  Torgersen        39.20000       19.60000         195.000000   
8       Adelie  Torgersen        34.10000       18.10000         193.000000   
..         ...        ...             ...            ...                ...   
334  Chinstrap      Dream        50.20000       18.80000         202.000000   
336  Chinstrap      Dream        51.90000       19.50000         206.000000   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   

     body_mass_g   sex  year  
0    3750.000000  male  2007  
3    4201.754386  male  2007  
5    3650.000000  male  2007  
7    4675.000000  male  2007  
8    3475.000000  male  2007  
..           ...   ...   ...  
334  3800.000000  male  2009  
336  3950.000000  male  2009  
339  4000.000000  male  2009  
341  3775.000000  male  2009  
342  4100.000000  male  2009  

[179 rows x 8 columns]

<h2>Insights#6
<h3>There are 165 Female penguins and Theres 179 Male Penguins

<h2>7.how many chinstrap are there in total

In [88]:
penguins [penguins ['species'] == 'Chinstrap']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

<h2>Insights#7
<h3>theres a total of 68 Chinstrap penguins